docs/DEVELOPER.md

```diff
@@ -186,3 +186,17 @@ A [tutorial on YouTube is available showing how to use XDebug and VSCode](https:
 Additional considerations:
 
 If you encounter errors such as VSCode cannot find a file in the path "vendor", these source files are not available to VSCode as they are running inside Docker.  If you want to debug vendor libraries (not required), you will have to use Composer to download dependencies and put them in your source tree.
+
```

Comment:
I'm just curious of the instructions on line 23 stopped working?  I don't know if ngrok is going to be a good way to go about this or not.

Reply:
So, testing here in the office demonstrates that this doesn't work out of the box.  My understanding is that a web server in docker running on localhost should not automatically be exposed outside of the dev machine unless the firewall had a port 80/443 hole explicitly poked in it.  On Windows I would imagine you have to create a Windows firewall rule to allow port 80/443 in from the outside.  In addition, I'm not sure if our office network is blocking port 80/443 between machines (unlikely) but I wasn't able to ask Tim E about it yet.

Reply:
I just followed the original instructions for accessing the app from a device an external device on the same network and it worked fine.  Let's meet some time and discuss the goal here and make sure I'm following everything that's being tried.

Reply:
So when you say it worked for you on your original device, are you saying you tested from your MacBook or do you have another machine you can spin up a dev environment on?  I am wondering if your machine is already "special" and has an open firewall whereas by default other machines do not.

I'll test at home as you asked for

Reply:
Yes, I spun things up on my MacBook and then accessed the so from my phone via my IP address

Reply:
I can confirm that this works on my Macbook Pro using Billy's instructions.  The reason this works, as far as I can tell, is because I am using Docker Desktop for Mac which has communicated with my Mac's firewall to poke the appropriate hole in the machine's firewall.  On both my and Lainey's machine, we are using WSL without Docker Desktop, which means that docker engine in ubuntu cannot automatically communicate with Windows Firewall to open the appropriate port automatically.

Some conclusions:
- we should give Docker Desktop for Windows and Linux another try to see if we can get away from using docker engine in Ubuntu.  This brings us closer to the same instructions for devs across platforms.
- we should put the ngrok instructions next to Billy's device testing section, as they are really two different ways to achieve the same thing.  Ngrok is really more for "worldwide access", which in theory would be nice to have Alex test out a prototype on my machine across the world but in practice we haven't needed it.

Reply:
@longrunningprocess I have combined the two device testing instructions into a single place at the bottom of the DEVELOPER readme.  IMHO both methods are valid and useful, and I'd like to keep them.  Now that I think we have a handle on why it wasn't working for @laineyhm or I on Windows, I'm satisfied with merging this.

Reply:
seems like a good idea for those local situations where using Desktop isn't feasible or for an extended use case like others accessing your local machine.